libmscore/instrchange.cpp

```diff
@@ -87,6 +88,19 @@ void InstrumentChange::read(XmlReader& e)
             else if (!Text::readProperties(e))
                   e.unknown();
             }
+      QString sv = score()->mscoreVersion();
+      sv.truncate(3);
+      if (sv == "2.0") {
```

Comment:
That part is easy enough to change without affecting anything else.  As mentioned, I have mixed feelings about that as well.  It maximizes "compatibility", but it's compatibility for a buggy behavior.  Removing that chunk of code would automatically fix some scores.